test/support/utils_complex_test_support.h

```diff
@@ -0,0 +1,178 @@
+// -*- C++ -*-
+//===----------------------------------------------------------------------===//
+//
+// Copyright (C) Intel Corporation
+//
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+// This file incorporates work covered by the following copyright and permission
+// notice:
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef __UTILS_COMPLEX_TEST_SUPPORT_H
+#define __UTILS_COMPLEX_TEST_SUPPORT_H
+
+#include <oneapi/dpl/execution>
+#include <oneapi/dpl/complex>
+
+#include "test_config.h"
+#include "utils.h"
+#include "utils_err_eng.h"
+
+#if TEST_DPCPP_BACKEND_PRESENT
+#    include <CL/sycl.hpp>
+#endif
+
+#include <oneapi/dpl/pstl/utils.h>
+
+// C++ compiler versions:
+//      C++11 : __cplusplus is 201103L
+//      C++14 : __cplusplus is 201402L
+//      C++20 : __cplusplus is 202002L
+
+namespace TestUtils
+{
+namespace Complex
+{
+    template <typename TComplexDataType>
+    struct InitConst;
+
+    template <>
+    struct InitConst<float>
+    {
+        using DestComplexFieldType = float;
+
+        static constexpr float kPartReal = 1.5f;
```

Comment:
why do we use such "complicated"  approach to have the initialization constants?

Reply:
I used this approach for consts initializations by different values depending on their type.

Reply:
I think it's appropriate, but I would rename the class template to e.g. `TestConstants` for better self-description.

Reply:
Fixed (renamed).